### PR TITLE
Light refresh of tests

### PR DESCRIPTION
**Pull request by jennybc**
This is a very light refresh of the tests that doesn't change them in substance. Will make a few comments in line.

BTW withr is a very lean dependency, which meetupr sort of already depends on anyway, because it's imported by testthat.

The use of `testthat::test_path()` also eliminates the dependency on here.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -18,10 +18,9 @@ Depends:
 Imports:
     httr,
     purrr,
-    tibble
-RoxygenNote: 6.1.1
+    tibble,
+    withr
```
Comment (jennybc) [inline DESCRIPTION:45]: New dependency, used in tests. I suppose it could also move to Suggests, alongside testthat.

File: README.Rmd [hunk 2]
```diff
@@ -80,12 +80,6 @@ Take a look at some resources:
 
 We are looking for people to help write tests and vignettes!  You can also take a look at the open [issues](https://github.com/rladies/meetupr/issues).
 
-In order to run our tests, you will have to set the `urlname` for meetup you belong to as an environment variable using the following code:
```
Comment (jennybc) [inline README.Rmd:83]: This doesn't seem to be true at the moment / anymore?

File: tests/testthat/test-find_groups.R [hunk 3]
```diff
@@ -1,13 +1,10 @@
-context("find_groups")
-
```
Comment (jennybc) [inline tests/testthat/test-find_groups.R:2]: testthat no longer needs `context()`s and they are sort of discouraged.

File: tests/testthat/test-find_groups.R [hunk 4]
```diff
@@ -1,13 +1,10 @@
...
 test_that("find_groups() success case", {
-
-  options('meetupr.use_oauth' = FALSE)
+  withr::local_options(list(meetupr.use_oauth = FALSE))
```
Comment (jennybc) [inline tests/testthat/test-find_groups.R:2]: Doing this via withr will make the change (set on option in this case) and then restore the previous state when the test exits. Overall this makes for fewer mysteries during development, i.e. the tests should leave state as they find it.

That is NOT true currently for the API key (several tests call `set_api_key()` without storing and restoring the original value), but I gather that's going away soon anyway, so I didn't touch it.

File: tests/testthat/test-find_groups.R [hunk 5]
```diff
@@ -1,13 +1,10 @@
...
   set_api_key("R-Ladies FTW!")
 
   meetup_groups <- with_mock(
     `httr::GET` = function(url, query, ...) {
-      load(here::here("tests/testdata/httr_get_find_groups.rda"))
+      load(test_path("testdata/httr_get_find_groups.rda"))
```
Comment (jennybc) [inline tests/testthat/test-find_groups.R:7]: `testthat::test_path()` is a more robust way to build this path. `here::here()` isn't going to work in all settings. Plus this switch removes the dependency on here.

File: tests/testthat/test-internals.R [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
   res <- with_mock(
     `httr::GET` = function(url, query, ...) {
-      load(here::here("tests/testdata/httr_get_find_groups.rda"))
+      load(test_path("testdata/httr_get_find_groups.rda"))
       return(req)
     },
-    res <- .quick_fetch(api_url = "fake url", # intentionally invalid as there is currently no validation
-                      api_key = "I <3 R-Ladies")
+    # intentionally invalid as there is currently no validation
+    res <- .quick_fetch(api_url = "fake url")
```
Comment (jennybc) [inline tests/testthat/test-internals.R:11]: This test didn't really seem to be *about* the API key? So I just set it above, like all the other tests, and now tests pass for me, without needing to set an env var.

File: tests/testthat/test-find_groups.R (hunk 5 above)
Comment (jennybc) [reply tests/testthat/test-find_groups.R:7]: I also moved the `.rda` files to put them in a place that's easy to access with `test_path()`.

File: DESCRIPTION (hunk 1 above)
Comment (ledell) [reply DESCRIPTION:45]: It's ok to add it.

**Review by ledell — APPROVED** [review]
Looks good, thanks!

File: README.Rmd (hunk 2 above)
Comment (ledell) [reply README.Rmd:83]: I don't recall what this was used for and it seems dated, so I am just going to say "yes, its not true".

**Review by ledell — APPROVED** [review]
This is a big improvement, thanks @jennybc!

**Pull request merged**